src/main/java/com/github/manolo8/darkbot/gui/drawables/DynamicEntitiesDrawer.java

```diff
@@ -133,21 +135,44 @@ private void drawNpcs(MapGraphics mg) {
     private void drawPlayers(MapGraphics mg) {
         for (Player player : players) {
             boolean isEnemy = player.getEntityInfo().isEnemy();
-            mg.setColor(isEnemy ? "enemies" : "allies");
-            drawEntity(mg, player, 4, false);
 
+            int level = -1;
+            String color = isEnemy ? "enemies" : "allies";
+            if (group.hasGroup()) {
+                GroupMember member = group.getMember(player);
+                if (member != null) {
+                    color = "group_member";
+                    level = member.getLevel();
+                }
+            }
+
+            mg.setColor(color);
+            drawEntity(mg, player, 4, false);
             if (mg.hasDisplayFlag(DisplayFlag.USERNAMES))
-                mg.drawString(player, player.getEntityInfo().getUsername(), -5, MapGraphics.StringAlign.MID);
+                mg.drawString(player, player.getEntityInfo().getUsername()
+                        + (level != -1 ? " [" + level + "]" : ""), -6, MapGraphics.StringAlign.MID);
+        }
+
+        if (group.hasGroup()) {
+            mg.setColor("group_member");
+            for (GroupMember member : group.getMembers()) {
+                if (member.getMapId() == hero.getMap().getId() && !isMemberVisible(member)) {
```

Comment:
I'm assuming this draws players who are not within render-distance, based on the coordinates gotten from group system.

If that's the case, i'd consider rendering these in a slightly different color/form. Maybe as a dotted outline, or a bit transparent, or a darker/lighter color, just something to signify there a bit of a "ghost" added-in by the bot. 
It would help know if they're within distance or not, as it would visually change in the ui whenever get got close/far from you.